### PR TITLE
sensord: lsm self test

### PR DESCRIPTION
**Pull request by gast04**
Add lsm6ds4 self test on sensord startup

### CODE REVIEW

File: selfdrive/sensord/sensors/lsm6ds3_accel.cc [hunk 1]
```diff
@@ -28,11 +144,29 @@ int LSM6DS3_Accel::init() {
     source = cereal::SensorEventData::SensorSource::LSM6DS3TRC;
   }
 
+  ret = perform_self_test(LSM6DS3_ACCEL_POSITIVE_TEST);
```
Comment (adeebshihadeh) [inline selfdrive/sensord/sensors/lsm6ds3_accel.cc:147]: let’s gate the self test behind an environment variable for this release. we should always run the test and log the result, but only assert if the environment variable is set. make sure to set it in the unit tests

File: selfdrive/sensord/sensors/lsm6ds3_accel.h [hunk 2]
```diff
@@ -10,21 +10,37 @@
...
+#define LSM6DS3_ACCEL_NEGATIVE_TEST   0b10
+#define LSM6DS3_ACCEL_MIN_ST_LIMIT_mg 90.0f
+#define LSM6DS3_ACCEL_MAX_ST_LIMIT_mg 1700.0f
 
 class LSM6DS3_Accel : public I2CSensor {
   uint8_t get_device_address() {return LSM6DS3_ACCEL_I2C_ADDR;}
   cereal::SensorEventData::SensorSource source = cereal::SensorEventData::SensorSource::LSM6DS3;
+
+  // self test functions
+  int perform_self_test(int test_type);
```
Comment (adeebshihadeh) [inline selfdrive/sensord/sensors/lsm6ds3_accel.h:41]: i’d just call this selftest

File: selfdrive/sensord/sensors/lsm6ds3_gyro.cc [hunk 3]
```diff
@@ -5,11 +5,113 @@
...
 LSM6DS3_Gyro::LSM6DS3_Gyro(I2CBus *bus, int gpio_nr, bool shared_gpio) : I2CSensor(bus, gpio_nr, shared_gpio) {}
 
+void LSM6DS3_Gyro::wait_for_data_ready() {
+  uint8_t drdy = 0;
+  uint8_t buffer[6];
+
+  do {
+    read_register(LSM6DS3_GYRO_I2C_REG_STAT_REG, &drdy, sizeof(drdy));
+    drdy &= LSM6DS3_GYRO_DRDY_GDA;
+  } while(drdy == 0);
```
Comment (adeebshihadeh) [inline selfdrive/sensord/sensors/lsm6ds3_gyro.cc:21]: make sure to be consistent with style
 ```suggestion
  } while (drdy == 0);
```

File: selfdrive/sensord/sensors/lsm6ds3_gyro.cc [hunk 4]
```diff
@@ -5,11 +5,113 @@
...
+
+  // read first values and discard
```
Comment (adeebshihadeh) [inline selfdrive/sensord/sensors/lsm6ds3_gyro.cc:23]: no needs for these kinds of comments. comments should explain things that aren’t immediately clear from reading the code

File: selfdrive/sensord/sensors/lsm6ds3_gyro.cc [hunk 5]
```diff
@@ -5,11 +5,113 @@
...
+        (test_val[i] > LSM6DS3_GYRO_MAX_ST_LIMIT_mdps)) {
+      test_result = false;
+    }
+  }
+
+  if (!test_result) {
+    ret = -1;
+  }
+
+fail:
```
Comment (adeebshihadeh) [inline selfdrive/sensord/sensors/lsm6ds3_gyro.cc:111]: usually goto’s are reserved for cases where you want to do some cleanup. in this case, i’d just return directly

File: selfdrive/sensord/sensors/lsm6ds3_accel.h (hunk 2 above)
Comment (gast04) [reply selfdrive/sensord/sensors/lsm6ds3_accel.h:41]: ok, will rename sticked to the one in bmx magn its called `perform_self_test` there as well

**Pull request merged**